src/weaponstats.cc

```diff
@@ -0,0 +1,171 @@
+#include "weaponstats.hh"
+#include "functions.hh"
+#include <cstdlib>
+#include <algorithm>
+#include "injector/injector.hpp"
+#include "logger.hh"
+#include "base.hh"
+#include <iostream>
+#include "config.hh"
+
+int                    statsStart                       = 0xC8AAB8;
+int                    itemSize                         = 0x70;
+WeaponStatsRandomizer *WeaponStatsRandomizer::mInstance = nullptr;
+int                    hexFlags[]
+    = {0x3033, 0x3833, 0x7001, 0x7011,  0x7031,  0x2001, 0x2011, 0x2031, 0x3013,
+       0x3033, 0xA008, 0xA014, 0x48214, 0x30238, 0x238,  0x0038, 0x0431};
+
+/*******************************************************/
+int __fastcall RandomizeStats (int *address, int weaponid)
+{
+    CPed *player = FindPlayerPed ();
+
+    if (address < (int *) player || address > (int *) (player + 0x764)
+        || weaponid == 41 || weaponid == 42 || weaponid == 43)
+        return 0;
+
+    int weaponids[] = {weaponid, 0, 0, 0};
+
+    if (weaponid + 25 >= 47 && weaponid + 25 <= 57)
+        {
+            weaponids[1] = weaponid + 25;
+            weaponids[2] = weaponid + 36;
+            weaponids[3] = weaponid + 47;
```

Comment:
Would be nice to have a comment explaining what this is for. As again, without checking the weapon id's I can't tell what this is meant to do.